source/fe/fe_enriched.cc

```diff
@@ -0,0 +1,949 @@
+
+  for (unsigned int i=0; i<fes.size(); i++)
+    if (multiplicities[i]>0)
+      this->base_to_block_indices.push_back( multiplicities[i] );
+
+  // TODO: we overload base_element() to use FESystem, probably not a big dealii
+  // to avoid setting up base_elements[] ?
+//  std::vector<Threads::Task<FiniteElement<dim,spacedim>*> > clone_base_elements;
+//
+//  for (unsigned int i=0; i<fes.size(); i++)
+//    if (multiplicities[i]>0)
+//      clone_base_elements.push_back (Threads::new_task (&FiniteElement<dim,spacedim>::clone,
+//                                                        *fes[i]));
+//
+//  unsigned int ind=0;
+//  for (unsigned int i=0; i<fes.size(); i++)
+//    {
+//      if (multiplicities[i]>0)
+//        {
+//          base_elements[ind] =
+//            std::make_pair (std_cxx11::shared_ptr<const FiniteElement<dim,spacedim> >
+//                            (clone_base_elements[ind].return_value()),
+//                            multiplicities[i]);
+//          ++ind;
+//        }
+//    }
+//
+//  Assert(ind>0, ExcInternalError());
+
+  {
+    // If the system is not primitive, these have not been initialized by
+    // FiniteElement
+    this->system_to_component_table.resize(this->dofs_per_cell);
+    this->face_system_to_component_table.resize(this->dofs_per_face);
+
+    FETools::Compositing::build_cell_tables(this->system_to_base_table,
+                                            this->system_to_component_table,
+                                            this->component_to_base_table,
+                                            *this,
+                                            false);
+
+    FETools::Compositing::build_face_tables(this->face_system_to_base_table,
+                                            this->face_system_to_component_table,
+                                            *this,
+                                            false);
+
+  }
+
+  // restriction and prolongation matrices are build on demand
+
+  // now set up the interface constraints for h-refinement.
+  // take them from fe_system:
+  this->interface_constraints = fe_system.interface_constraints;
+
+  // if we just wrap another FE (i.e. use FE_Nothing as a second FE)
+  // then it makes sense to have support points.
+  // However, functions like interpolate_boundary_values() need all FEs inside
+  // FECollection to be able to provide support points irrespectively whether
+  // this FE sits on the boundary or not. Thus for moment just copy support
+  // points from fe system:
+  {
+    this->unit_support_points      = fe_system.unit_support_points;
+    this->unit_face_support_points = fe_system.unit_face_support_points;
+  }
+
+  // take adjust_quad_dof_index_for_face_orientation_table from FESystem:
+  {
+    this->adjust_line_dof_index_for_line_orientation_table = fe_system.adjust_line_dof_index_for_line_orientation_table;
+  }
+}
+
+
+template <int dim, int spacedim>
+FE_Enriched<dim,spacedim>::~FE_Enriched ()
+{
+}
```

Comment:
Because the destructor of the base class is `virtual`, the one of the current class is also `virtual` -- whether you actually declare/define it or not.